index.js

```diff
@@ -88,7 +90,12 @@ AssetsWebpackPlugin.prototype = {
         output.metadata = self.options.metadata
       }
 
-      self.writer(output, function (err) {
+      if (!compiler.outputFileSystem.readFile) {
+        compiler.outputFileSystem.readFile = fs.readFile.bind(fs);
+        compiler.outputFileSystem.join = path.join.bind(path);
```

Comment:
You'll need to remove the semicolons on 94-95 to pass lint, otherwise looks good.